mips2c_macros.h

```diff
@@ -0,0 +1,17 @@
+#ifndef _MIPS2C_MACROS_H_
+#define _MIPS2C_MACROS_H_
+
+// Unknown types
+#define MIPS2C_UNK      s32
+#define MIPS2C_UNK8     s8
+#define MIPS2C_UNK16    s16
+#define MIPS2C_UNK32    s32
+#define MIPS2C_UNK64    s64
+
+// Bitwise (reinterpret) cast
+#define MIPS2C_BITWISE(_type, _expr) (*(_type *)&(_expr))
+
+// Unknown field access, like `(_type) _expr->unk_offset`
+#define MIPS2C_FIELD(_expr, _type, _offset) (*(_type *) ((s8 *)(_expr) + (_offset)))
+
+#endif
```

Comment:
`#define ERROR(x) (0)` would be another thing we could add here; it's blatantly wrong but may cause some misidentified float -> unsigned conversions to compile

Reply:
Yeah good call, I'll also add `TRAP_IF`, `BREAK`, and `SYNC`. 
I think it's OK if they're always prefixed with `MIPS2C_`? (e.g `MIPS2C_ERROR`?) If not that should be easy to fix.

Reply:
Sure, that sounds okay. TRAP_IF and SYNC won't come up in practice, they are restricted to handwritten asm. But BREAK may, for misidentified divisions/modulo.